cloudsync/tasks.py

```diff
@@ -23,58 +26,92 @@
 THUMBNAIL_PATTERN = "thumbnails/{}_thumbnail_{{count}}"
 
 
-@shared_task(bind=True)
-def stream_to_s3(self, url, s3_key):
+class VideoTask(Task):
     """
-    Stream the contents of the given URL to Amazon S3
+    Custom Celery Task class for video uploads and transcodes
     """
-    if not url:
-        return False
-    response = requests.get(url, stream=True)
-    response.raise_for_status()
-    _, content_type, content_length = parse_content_metadata(response)
 
-    s3 = boto3.resource('s3')
-    bucket_name = settings.VIDEO_S3_BUCKET
-    bucket = s3.Bucket(bucket_name)
+    def get_task_id(self):
+        """
+        Get the task id (depending on whether the task is chained or not)
 
-    # Need to bind this here, because otherwise it gets lost in the callback somehow
-    task_id = self.request.id
+        Args:
+            request(Task.request): The task request
 
-    def callback(bytes_uploaded):
+        Returns:
+            The request id
         """
-        Callback function after upload
-        """
-        data = {
-            "uploaded": bytes_uploaded,
-            "total": content_length,
-        }
-        self.update_state(task_id=task_id, state="PROGRESS", meta=data)
-
-    config = TransferConfig(**settings.AWS_S3_UPLOAD_TRANSFER_CONFIG)
-
-    bucket.upload_fileobj(
-        Fileobj=response.raw,
-        Key=s3_key,
-        ExtraArgs={"ContentType": content_type},
-        Callback=callback,
-        Config=config
-    )
+        if self.request.chain:
+            return self.request.chain[0]['options']['task_id']
```

Comment:
ok then
but please wrap the call with a try/except so you can log what went wrong in case the structure changes.
Also you should write tests for this